ui-app/client/src/components/util/getPositions.ts

```diff
@@ -0,0 +1,116 @@
+export default function getPositions(position: String, boxRect: DOMRect, popoverRect: DOMRect) {
```

Comment:
The position data type has to be an enum and create an enum here and use it in the component.